### PR TITLE
Removing non-quant code and data

### PR DESCRIPTION
**Pull request by nickmalleson**
This PR deals with issue https://github.com/Urban-Analytics/RAMP-UA/issues/255 , removal of the old, non-quant code and data. There are some minor things that have been noted in the discussion about the issue https://github.com/Urban-Analytics/RAMP-UA/issues/255#issuecomment-792907978 (no need to repeat them here)

Main changes:

- Removed the code in `population_initialisation.py` that checked for 'non-quant' mode and ran the devon-specific shop and school allocation
- The `quant_object` is no longer class-level parameter and instead is passed directly to `population_initialisation` constructor and then on to the functions that need it
- Removed the `quant-dir` parameter in the default parameters file
https://github.com/Urban-Analytics/RAMP-UA/blob/445649fa018e1348e423e3945b74054864752f97/model_parameters/default.yml#L12
- Updated some tests and removed the tests that checked the Devon-specific files had been read correctly.

### CODE REVIEW

**Comment by rory-improbable** [comment]
> @rory-improbable when you get a minute (not urgent) would you mind having a look at this for me? It relates to changes I am making on branch [nickmalleson-issue-255](https://github.com/Urban-Analytics/RAMP-UA/tree/nickmalleson-issue-255) I have removed some old code that read in bespoke shopping and retail venues for Devon and everything seems to work OK when I run it, but the OpenCL tests are failing and I can't work out why (error below).
> 
> The only change that springs to mind is to the `test_update_flows_kernel.py` file: https://github.com/Urban-Analytics/RAMP-UA/blob/7a6b5e1bf44f112fb48b7b4917527106b012bedf/tests/opencl/test_update_flows_kernel.py . I needed to add the `quant_object` argument when conducting the population initialisation. But I am not sure why this would break the test. Any ideas? As I said the OpenCL model seems to run fine. Thanks!
> 
> ```
> Testing started at 13:29 ...
> /Users/nick/opt/anaconda3/envs/ramp-ua/bin/python /Applications/PyCharm.app/Contents/plugins/python/helpers/pycharm/_jb_pytest_runner.py --path /Users/nick/research_not_syncd/git_projects/RAMP-UA/tests -- --last-failed
> Launching pytest with arguments --last-failed /Users/nick/research_not_syncd/git_projects/RAMP-UA/tests in /Users/nick/research_not_syncd/git_projects/RAMP-UA
> 
> ============================= test session starts ==============================
> platform darwin -- Python 3.7.10, pytest-5.4.1, py-1.10.0, pluggy-0.13.1 -- /Users/nick/opt/anaconda3/envs/ramp-ua/bin/python
> cachedir: .pytest_cache
> rootdir: /Users/nick/research_not_syncd/git_projects/RAMP-UA
> collecting ... collected 6 items
> run-last-failure: rerun previous 6 failures (skipped 9 files)
> 
> tests/opencl/test_snapshot_converter.py::test_global_id_lookup 
> tests/opencl/test_snapshot_converter.py::test_processes_people_flows 
> tests/opencl/test_snapshot_converter.py::test_get_place_data 
> tests/opencl/test_snapshot_converter.py::test_get_coordinates 
> Converting Home flows for all people: 100%|██████████| 3/3 [00:00<00:00, 17747.41it/s]
> Converting Retail flows for all people: 100%|██████████| 3/3 [00:00<00:00, 15553.66it/s]
> Storing location type for Home: 100%|██████████| 3/3 [00:00<00:00, 35746.91it/s]
> Storing location type for Retail: 100%|██████████| 5/5 [00:00<00:00, 54899.27it/s]
> Processing coordinate data for Retail: 5it [00:00, 8878.71it/s]
> Storing coordinates for homes: 3it [00:00, 35246.25it/s]
> Fatal Python error: Aborted
> 
> Thread 0x000070000525c000 (most recent call first):
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/threading.py", line 300 in wait
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/threading.py", line 552 in wait
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/tqdm/_monitor.py", line 60 in run
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/threading.py", line 926 in _bootstrap_inner
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/threading.py", line 890 in _bootstrap
> 
> Thread 0x0000000114adbe00 (most recent call first):
>   File "/Users/nick/research_not_syncd/git_projects/RAMP-UA/microsim/opencl/ramp/simulator.py", line 191 in step_kernel
>   File "/Users/nick/research_not_syncd/git_projects/RAMP-UA/tests/opencl/test_update_flows_kernel.py", line 72 in test_correct_flow_calculation_no_lockdown
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/_pytest/python.py", line 184 in pytest_pyfunc_call
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/_pytest/python.py", line 1479 in runtest
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/_pytest/runner.py", line 135 in pytest_runtest_call
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/_pytest/runner.py", line 217 in <lambda>
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/_pytest/runner.py", line 244 in from_call
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/_pytest/runner.py", line 217 in call_runtest_hook
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/_pytest/runner.py", line 186 in call_and_report
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/_pytest/runner.py", line 100 in runtestprotocol
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/_pytest/runner.py", line 85 in pytest_runtest_protocol
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/_pytest/main.py", line 272 in pytest_runtestloop
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/_pytest/main.py", line 247 in _main
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/_pytest/main.py", line 191 in wrap_session
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/_pytest/main.py", line 240 in pytest_cmdline_main
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
>   File "/Users/nick/opt/anaconda3/envs/ramp-ua/lib/python3.7/site-packages/_pytest/config/__init__.py", line 125 in main
>   File "/Applications/PyCharm.app/Contents/plugins/python/helpers/pycharm/_jb_pytest_runner.py", line 43 in <module>
> 
> Process finished with exit code 134 (interrupted by signal 6: SIGABRT)
> ```

hey @nickmalleson I can definitely have a look at this. The tests are a bit brittle so I'm sure it's something simple that is breaking them.

**Comment by rory-improbable** [comment]
@nickmalleson so I've checked out this branch and all the tests are actually passing for me locally now (once I downloaded the devon_data folder). Is this the same situation that you have where the tests are passing locally but not in CI? or are they failing for you locally?

**Comment by rory-improbable** [comment]
OK I think the problem may be that the CI agent running the tests doesn't have the devon data downloaded. 

It seems like only the MacOS agent is failing, the tests seem to pass on Ubuntu.

This is the error from the MacOS agent, it's the same error I got when I didn't have the devon_data directory downloaded

` ==================================== ERRORS ====================================
________________ ERROR collecting tests/test_microsim_model.py _________________
tests/test_microsim_model.py:19: in <module>
    "quant_object": QuantRampAPI(os.path.join("devon_data", "QUANT_RAMP"))
microsim/quant_api.py:27: in __init__
    QuantRampAPI.read_data(self.QUANT_DIR)
microsim/quant_api.py:35: in read_data
    cls.dfPrimaryPopulation = pd.read_csv(os.path.join(QUANT_DIR,'primaryPopulation.csv'))
/usr/local/miniconda/envs/ramp-ua/lib/python3.7/site-packages/pandas/io/parsers.py:676: in parser_f
    return _read(filepath_or_buffer, kwds)
/usr/local/miniconda/envs/ramp-ua/lib/python3.7/site-packages/pandas/io/parsers.py:448: in _read
    parser = TextFileReader(fp_or_buf, **kwds)
/usr/local/miniconda/envs/ramp-ua/lib/python3.7/site-packages/pandas/io/parsers.py:880: in __init__
    self._make_engine(self.engine)
/usr/local/miniconda/envs/ramp-ua/lib/python3.7/site-packages/pandas/io/parsers.py:1114: in _make_engine
    self._engine = CParserWrapper(self.f, **self.options)
/usr/local/miniconda/envs/ramp-ua/lib/python3.7/site-packages/pandas/io/parsers.py:1891: in __init__
    self._reader = parsers.TextReader(src, **kwds)
pandas/_libs/parsers.pyx:374: in pandas._libs.parsers.TextReader.__cinit__
    ???
pandas/_libs/parsers.pyx:674: in pandas._libs.parsers.TextReader._setup_parser_source
    ???
E   FileNotFoundError: [Errno 2] File devon_data/QUANT_RAMP/primaryPopulation.csv does not exist: 'devon_data/QUANT_RAMP/primaryPopulation.csv'`

**Comment by nickmalleson** [comment]
Thanks @rory-improbable I think we're getting there. I hadn't thought about the data being an issue. This is failing for me locally (on a mac) but the data may well be a data issue. @Sparrow0hawk and I can discuss on Thursday. Enjoy your holiday.

**Comment by nickmalleson** [comment]
The error above doesn't occur when running `pytest` from the command-line, so lets ignore it. The online tests will fail though because the required QUANT data aren't included as part of the tests. Will deal with this in https://github.com/Urban-Analytics/RAMP-UA/issues/267

**Comment by Sparrow0hawk** [comment]
Just want to consolidate what needs to be done for this PR (@nickmalleson) let me know if i've missed anything:
- [ ] - tests for quant_api [#262]
- [ ]  - adding dummy data for quant_api [#267]
- [ ] - tidying up docstrings (working on in [quant_api-tidy](https://github.com/Urban-Analytics/RAMP-UA/tree/quant_api-tidy) branch)

**Comment by nickmalleson** [comment]
@Sparrow0hawk I think that this one is part of the PR:

 * [ ]   - adding dummy data for quant_api [#267]

But two these aren't: 


>    * [ ]  - tests for quant_api [#262]

  - This is the subject of  https://github.com/Urban-Analytics/RAMP-UA/issues/262 



>    * [ ]  - tidying up docstrings (working on in [quant_api-tidy](https://github.com/Urban-Analytics/RAMP-UA/tree/quant_api-tidy) branch)

  - This is the subject of: https://github.com/Urban-Analytics/RAMP-UA/issues/257

**Comment by nickmalleson** [comment]
@Sparrow0hawk I think this is good to go. Do you want to have a quick look through? I don't have anything in particular that I think you should look at.

**Comment by nickmalleson** [comment]
Made one more minor change. Rather than having a `test_mode` that created small flow matrices, I just wrote out the small matricies and included them in the dummy data so that the behaviour of quant in the tests is exactly the same as it is under normal conditions)

Note, code to generate the matrices:
```
pickle.dump(np.tile(np.array([0.001] * 8), (8, 1)), open("FILENAME.bin", 'wb'))
```

File: tests/dummy_data/QUANT_RAMP/add_msoa_to_venue.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
```
Comment (Sparrow0hawk) [inline tests/dummy_data/QUANT_RAMP/add_msoa_to_venue.py:1]: Is this file in the most appropriate place? Is it more an utilities script?

And could we include some documentation about what this script is for?

File: tests/dummy_data/QUANT_RAMP/filter_data2.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# After running filter_data2 to find out which areas are included in the test data,
```
Comment (Sparrow0hawk) [inline tests/dummy_data/QUANT_RAMP/filter_data2.py:1]: As with above, unsure about the right place for this script.

File: tests/dummy_data/QUANT_RAMP/primaryPopulation.csv [hunk 3]
```diff
@@ -0,0 +1,9 @@
+,Unnamed: 0,rowid,msoaiz,count_primary,count_secondary,zonei,lat,lon,osgb36_east,osgb36_north,area
```
Comment (Sparrow0hawk) [inline tests/dummy_data/QUANT_RAMP/primaryPopulation.csv:1]: This file has an `Unnamed: 0` column which is typically created when using `pd.to_csv()` but not setting `index=False` to preventing it writing out the index column as a new column. Should just be careful as regular `read_csv` and `to_csv` calls will continue to create lots of Unnamed columns.

File: tests/dummy_data/QUANT_RAMP/secondaryPopulation.csv [hunk 4]
```diff
@@ -0,0 +1,9 @@
+,Unnamed: 0,rowid,msoaiz,count_primary,count_secondary,zonei,lat,lon,osgb36_east,osgb36_north,area
```
Comment (Sparrow0hawk) [inline tests/dummy_data/QUANT_RAMP/secondaryPopulation.csv:1]: Same as comment on `primaryPopulation.csv`

File: tests/dummy_data/devon-tu_health/Devon_simulated_TU_keyworker_health.csv [hunk 5]
```diff
@@ -1,18 +1,18 @@
...
-E02004151,5,23968,1,5,70,Health professionals,0,0,0,0.4201389,,,0,0,0,0,0,0,0,0,0,0,0
-E02004159,8,90653,1,3,40,"Leisure, travel and related personal service occupations",0,0,0,,,0.086805552,0,0,0,0,0,0,0,0,0,0,0
-E02004142,0,100799,0,4,50,Health professionals,0,0,0,,,0.052083332,0,0,0,0,0,0,0,0,0,0,0
-E02004132,1,21179,1,2,18,"Skilled metal, electrical and electronic trades",0,0,0,,0.27777779,,0,0,0,0,0,0,0,0,0,0,0
-E02004132,1,25604,1,3,29,"Leisure, travel and related personal service occupations",0,0,0,,,,0,0,0,0,0,0,0,0,0,0,0
-E02004147,3,132614,0,4,55,Elementary administration and service occupations,0,0,0,0.2361111,,,0,0,0,0,0,0,0,0,0,0,0
-E02004147,3,133530,0,2,15,Not applicable,0,0,0,,,,0,0,0,0,0,0,0,0,0,0,0
-E02004147,3,137456,1,5,58,Health professionals,0,0,0,,,0.086805552,0,0,0,0,0,0,0,0,0,0,0
-E02004138,6,69652,1,3,43,"Leisure, travel and related personal service occupations",0,0,0,0.2361111,,,0,0,0,0,0,0,0,0,0,0,0
+﻿area,old_area,hid,pid,Sex,Age1,age,soc2010,punknown,phome,pworkhome,pwork,pschool,pshop,pservices,pleisure,pescort,ptransport,pnothome,phometot,pmwalk,pmcycle,pmprivate,pmpublic,pmunknown
```
Comment (Sparrow0hawk) [inline tests/dummy_data/devon-tu_health/Devon_simulated_TU_keyworker_health.csv:1]: is `old_area` required?

File: tests/test_microsim_model.py [hunk 6]
```diff
@@ -27,7 +31,8 @@
 # like `test_step()`.
 @pytest.fixture()
 def test_microsim():
-    population_init = PopulationInitialisation(**population_init_args)
+    with pytest.warns(UserWarning):  # (PopInit should throw a warning because the QUANT api is in test mode
```
Comment (Sparrow0hawk) [inline tests/test_microsim_model.py:34]: Think you've removed the test_mode so this change maybe unnecessary

File: tests/test_population_initialisation.py [hunk 7]
```diff
@@ -86,8 +89,14 @@ def test_population_init():
             venue = workplaces.iloc[venue_number]
             assert venue['SOC'] == population_init.individuals.at[p, "soc2010"]
         assert set(workplaces.loc[population_init.individuals.at[p, "Work_Venues"], "MSOA"]) == \
-            set(['E02004158', 'E02004147', 'E02004132', 'E02004138', 'E02004159'])
+            set(['E02004290', 'E02004301', 'E02004292', 'E02004300', 'E02004299'])
 
+    # TODO Test shops and school allocations.
```
Comment (Sparrow0hawk) [inline tests/test_population_initialisation.py:94]: Should this be an issue if it's to be addressed in a subsequent PR?

**Review by Sparrow0hawk — CHANGES_REQUESTED** [review]
All looks good. I've made some general suggestions in the code. 

All models still run for me, and tests pass for me locally.

We still need some tests for the quant stuff but perhaps we can borrow these directly from https://github.com/maptube/QUANT_RAMP

There's a few scripts you've included in the `tests/dummy_data` directory which may be better in `utilities` or a new directory in tests specifically for scripts around test data generation.

File: tests/dummy_data/QUANT_RAMP/add_msoa_to_venue.py [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+        pointInPolys = sjoin(gdf_venues[key], poly, how='left')
+        grouped = pointInPolys.groupby('index_right')
+        results = list(grouped)
+        if results: # at least one venue
+            tmp = results[0][1]
+            venues[key].iloc[tmp.zonei,4] = m
+    venues[key].to_csv (os.path.join(base_dir,"data","QUANT_RAMP","model-runs",f"test{key}.csv"), index = True, header=True)
+
+
+# import matplotlib.pyplot as plt
```
[the remark this answers is not in the document]
Comment (nickmalleson) [reply tests/dummy_data/QUANT_RAMP/add_msoa_to_venue.py:83]: Done

File: tests/dummy_data/QUANT_RAMP/filter_data2.py [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+        if any(zone == int(line_array[0]) for zone in INPUT_ZONES) :
+            print(line.strip())
+            matched_lines += 1
+
+assert matched_lines == len(INPUT_ZONES), f"Warning, there are {len(INPUT_ZONES)} to look for, but only {matched_lines} lines found"
+print("Finished")
+
+
+"""
```
[the remark this answers is not in the document]
Comment (nickmalleson) [reply tests/dummy_data/QUANT_RAMP/filter_data2.py]: No, good call, removed

File: tests/dummy_data/QUANT_RAMP/primaryPopulation.csv (hunk 3 above)
Comment (nickmalleson) [reply tests/dummy_data/QUANT_RAMP/primaryPopulation.csv:1]: Good point. I left it in as I wanted it to be identical to the main data used, and am loath to change it now, but will be aware in future.

File: tests/test_microsim_model.py (hunk 6 above)
Comment (nickmalleson) [reply tests/test_microsim_model.py:34]: Well spotted, removed

File: tests/dummy_data/devon-tu_health/Devon_simulated_TU_keyworker_health.csv (hunk 5 above)
Comment (nickmalleson) [reply tests/dummy_data/devon-tu_health/Devon_simulated_TU_keyworker_health.csv:1]: No, but lets keep it in incase there's a need to refer back to the old, Devon, area names for some reason

File: tests/dummy_data/QUANT_RAMP/add_msoa_to_venue.py (hunk 1 above)
Comment (nickmalleson) [reply tests/dummy_data/QUANT_RAMP/add_msoa_to_venue.py:1]: Sorry, I've no idea! I suggest we leave it

**Comment by nickmalleson** [comment]
> There's a few scripts you've included in the `tests/dummy_data` directory which may be better in `utilities` or a new directory in tests specifically for scripts around test data generation.

I used these to hack together the test data. We don't need them, but I think it's worth keeping them in case we want to re-create the test data again. It was a painful process. I thnk they're better where they are rather than in `utilities`, as putting them there implies that they have some useful function? Which they don't any more. But am happy with your choice.

**Comment by Sparrow0hawk** [comment]
> 
> 
> > There's a few scripts you've included in the `tests/dummy_data` directory which may be better in `utilities` or a new directory in tests specifically for scripts around test data generation.
> 
> I used these to hack together the test data. We don't need them, but I think it's worth keeping them in case we want to re-create the test data again. It was a painful process. I thnk they're better where they are rather than in `utilities`, as putting them there implies that they have some useful function? Which they don't any more. But am happy with your choice.

True, perhaps worth leaving a short `README.md` in this directory just to give a summary of what these files are for and brief guidance of how they work?

**Review by Sparrow0hawk — APPROVED** [review]

**Pull request merged**